crates/bevy_asset/src/path.rs

```diff
@@ -465,6 +465,38 @@ impl<'a> AssetPath<'a> {
         Some(extension)
     }
 
+    /// Returns the path of the [`AssetPath`] relative to the app root.
+    /// Returns None if the [`AssetPath`] uses a non-default Asset Source.
+    ///
+    /// Ex: an [`AssetPath`] referencing "/models/apple.gltf" would return a [`PathBuf`] for "assets/models/apple.gltf"
+    pub fn get_root_relative_path(&self) -> Option<PathBuf> {
+        if let AssetSourceId::Name(_) = self.source {
+            return None;
+        }
+        let path = self.path();
+        let path = if path.is_absolute() {
+            match path.strip_prefix("/") {
+                Ok(path) => path,
+                Err(_) => {
+                    return None;
+                }
+            }
+        } else {
+            path
+        };
```

Comment:
Or the code-golf version xD not sure which I prefer

```suggestion
        let path = self.path();
        let path = path.is_absolute().then(|| path.strip_prefix("/").ok()).flatten();
```